lib/gscraper/accounts/user.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Gscraper.Accounts.User do
+  use Ecto.Schema
+  import Ecto.Changeset
+  import GscraperWeb.Gettext
+
+  alias Gscraper.Accounts.Password
+
+  schema "users" do
+    field :username, :string
+    field :encrypted_password, :string
+
+    field :password, :string, virtual: true
+    field :password_confirmation, :string, virtual: true
+
+    timestamps()
+  end
+
+  @fields ~w(username password password_confirmation)a
+
+  @doc false
+  def changeset(user, attrs) do
+    user
+    |> cast(attrs, @fields)
+    |> validate_required(@fields)
+    |> unique_constraint(:username)
+    |> validate_password
+  end
+
+  def validate_password(changeset) do
```

Comment:
Maybe rename this to be describe more about the function, `validate_and_hash_password`

Reply:
I moved the method call to "validate_password" to the pipeline of the `changeset` method in 6032768. It makes more sense to put it there, I think.